updater/internal/install/install.go

```diff
@@ -29,9 +29,16 @@ import (
 	"go.uber.org/zap"
 )
 
-// Installer allows you to install files from latestDir into installDir,
+//Installer is an interface that performs an Install of a new collector.
+//go:generate mockery --name Installer --filename installer.go
+type Installer interface {
+	// Install installs new artifacts over the old ones.
+	Install(rollback.Rollbacker) error
+}
+
+// installer allows you to install files from latestDir into installDir,
 // as well as update the service configuration using the "Install" method.
-type Installer struct {
+type installer struct {
```

Comment:
nit: I'm not a huge fan of having the implementation of the interface be lower case. Feels like this could lead to confusion. Maybe even just calling this `packageInstaller` to differentiate it from the interface a bit more.